registry/tests/api/main.rs

```diff
@@ -23,30 +31,65 @@ pub fn create_publish_request_sample() -> PublishRequest {
     }
 }
 
-#[tokio::test]
-async fn test_publish_registry() -> Result<(), Box<dyn std::error::Error>> {
+pub async fn basic_setup() -> Result<RegistryClient<Channel>, Box<dyn std::error::Error>> {
```

Comment:
Two things here:

1. Box<dyn std::error::Error>> is rarely a good idea to use directly in your code base. There are two options: a) use `eyre`/`miette` or b) use concrete error enums to cover for all your cases. I think eyre is the most appropriate here
2. Stuff like this (which you want to "paste" into many functions") could/would be a macro ideally, this also frees you from certain drop restrictions (making it a function works nonetheless, its more of a nit/style question)

Reply:
PS: as we are in tests, its ok to use `unwrap` or `expect` to skip error handling entirely, this would simplify your types a lot